### DISCUSSION TITLE
Convergence Criterion

### DISCUSSION THREAD

**Discussion by Krystalfang** [idaholab/moose#21653] [Q&A General] [answered]
Dear Expert,
I recently tried to use MOOSE to simulate the conjugate heat transfer, but encountered the problem of non-convergence, so I want to ask what is the convergence criterion of MOOSE,  I set nl_rel_tol = 1e-6.

![LMV$B~UIJUUI(YV$ARVJQ)1](https://user-images.githubusercontent.com/107802065/180113042-a3a89883-c6f3-46bd-bbd0-f23bd7e7734c.png)

But it has been calculated until 10^-7, and the residuals of each variable are also less than 10^-6. The result is still not convergent, why is this?

![29)M43GJG%LBY5LAM@E94@2](https://user-images.githubusercontent.com/107802065/180113020-ca8ecd23-dd8f-4dd2-be78-0ec93aa3a37b.png)

![{G XCMET I3%_K3AOB%%B09](https://user-images.githubusercontent.com/107802065/180113027-7c153132-3547-433b-942d-cde07378c31a.png)

**Comment by dschwen** [comment, marked as answer]
You are mixing absolute and relative tolerance up. Try setting `nl_abs_tol = 1e-6`. Your setting requires the initial residual to drop by 6 orders of magnitude, which is not always possible, if you start out with an initial guess that is very close to the solution.

**Comment by GiudGiud** [comment]
The default convergence criteria are listed as the default parameters to the Executioner here:
https://mooseframework.inl.gov/source/executioners/Steady.html

Please post a full log rather than these screenshots so we can see what is going on with the convergence history
